source/Plugins.cpp

```diff
@@ -0,0 +1,135 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	// The first item in pair is plugin state on load, second item is user changing preference.
+	map<string, pair<bool, bool>> settings;
```

Comment:
You can use an anonymous class type if you would rather not deal with `.second.first` and `.second.second`.

E.g.
```c++
struct EnabledState {
  bool initial = true;
  bool current = true;
}
map<string, EnabledState> settings;
```

Reply:
Can you elaborate how this would be used?  I see the definition but not clear to me how I would instantiate it or use it.

Reply:
I added the `EnabledState` struct.  I figured out how to use it once I was in an IDE which gave me hints on how to use it.

I also added a constructor which allows me to customize the `initial` and `current` state which gets used on GameData loading of `plugins.txt`.